src/openrct2/windows/map.c

```diff
@@ -1349,41 +1347,32 @@ static void window_map_place_park_entrance_tool_down(sint32 x, sint32 y)
  */
 static void window_map_set_peep_spawn_tool_down(sint32 x, sint32 y)
 {
-    rct_map_element *mapElement, *surfaceMapElement;
+    rct_map_element *mapElement;
     sint32 mapX, mapY, mapZ, direction;
 
     footpath_get_coordinates_from_pos(x, y, &mapX, &mapY, &direction, &mapElement);
-    if (mapX == 0x8000)
-        return;
-
-    surfaceMapElement = map_get_surface_element_at(mapX >> 5, mapY >> 5);
-    if (surfaceMapElement == NULL) {
-        return;
-    }
-    if (surfaceMapElement->properties.surface.ownership & 0xF0) {
-        return;
-    }
-
-    mapX = mapX + 16 + (word_981D6C[direction].x * 15);
-    mapY = mapY + 16 + (word_981D6C[direction].y * 15);
```

Comment:
I'm not sure how this is happening, but 
```
if (mapX != MAP_LOCATION_NULL) {
sint32 mapXOld, mapYOld;
mapXOld = mapX + 16 + (word_981D6C[direction].x * 15);
mapYOld = mapY + 16 + (word_981D6C[direction].y * 15);
assert(mapX == mapXOld && mapY == mapYOld);
mapZ = mapElement->base_height / 2;
```
Is not triggering any assertion errors. At the same time, having.
```
if (mapX != MAP_LOCATION_NULL) {
mapX = mapX + 16 + (word_981D6C[direction].x * 15);
mapY = mapY + 16 + (word_981D6C[direction].y * 15);
mapZ = mapElement->base_height / 2;
```
Causes the placement to not work. I'm not sure how this is happening, it doesn't make sense to me. Can I push the version with the assert comparison and you guys tell me if you can trigger the assert?